src/tlo/scenario.py

```diff
@@ -125,6 +127,43 @@ def __init__(
         self.resources = resources_path
         self.rng = None
         self.scenario_path = None
+        self.arguments = None
+
+    def parse_arguments(self, extra_arguments: List[str]) -> None:
+        """Base class command line arguments handling for scenarios. This should not be overridden by subclasses.
+        Subclasses can add argument handling to their classes by implementing the `add_arguments` method."""
+
+        self.arguments = extra_arguments
+
+        parser = argparse.ArgumentParser()
+        parser.add_argument("--resume-simulation", type=str, help="Resume simulation from directory")
+        parser.add_argument("--suspend-date", type=str_to_pandas_date, help="Suspend the simulation")
+
+        # add arguments from the subclass
+        self.add_arguments(parser)
+
+        arguments = parser.parse_args(self.arguments)
+
+        # set the arguments as attributes of the scenario
+        for key, value in vars(arguments).items():
+            if value is not None:
+                if hasattr(self, key):
+                    logger.info(key="message", data=f"Overriding attribute: {key}: {getattr(self, key)} -> {value}")
+                setattr(self, key, value)
```

Comment:
This assumes we never want an argument to have value `None` - this is probably fine but we might want to document this in the `add_arguments` docstring?